pkg/ui/admin/controller/namespaces.go

```diff
@@ -0,0 +1,127 @@
+package controller
+
+import (
+	"github.com/gofiber/fiber/v2"
+
+	"github.com/G-Research/fasttrackml/pkg/ui/admin/request"
+	"github.com/G-Research/fasttrackml/pkg/ui/admin/response"
+)
+
+// GetNamespaces renders the list view with no message.
+func (c Controller) GetNamespaces(ctx *fiber.Ctx) error {
+	return c.renderIndex(ctx, "")
+}
+
+// GetNamespace renders the update view for a namespace.
+func (c Controller) GetNamespace(ctx *fiber.Ctx) error {
+	p := struct {
+		ID uint `params:"id"`
+	}{}
+	if err := ctx.ParamsParser(&p); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+	ns, err := c.namespaceService.GetNamespace(ctx.Context(), p.ID)
+	if err != nil {
+		return fiber.NewError(fiber.ErrInternalServerError.Code, "unable to find namespace")
+	}
+	if ns == nil {
+		return fiber.NewError(fiber.StatusNotFound, "namespace not found")
+	}
+	return ctx.Render("ns/update", fiber.Map{
+		"ID":          ns.ID,
+		"Code":        ns.Code,
+		"Description": ns.Description,
+	})
+}
+
+// NewNamespace renders the create view for a namespace.
+func (c Controller) NewNamespace(ctx *fiber.Ctx) error {
+	ns := response.Namespace{}
+	return ctx.Render("ns/create", fiber.Map{
```

Comment:
if there is an error, the form needs a backing object to display the previous data. so that means an empty object is provided on the first request for consistency.